src/components/diagrams/diagram-layout.tsx

```diff
@@ -0,0 +1,313 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useCallback, useState } from 'react';
+import { Layout, Layouts, Responsive, WidthProvider } from 'react-grid-layout';
+import { useDiagramModel } from './hooks/use-diagram-model';
+import { Diagram, DiagramParams, DiagramType } from './diagram.type';
+import { Box, darken, IconButton, Theme, useTheme } from '@mui/material';
+import { EquipmentInfos, EquipmentType, OverflowableText } from '@gridsuite/commons-ui';
+import CloseIcon from '@mui/icons-material/Close';
+import LibraryAddOutlinedIcon from '@mui/icons-material/LibraryAddOutlined';
+import { UUID } from 'crypto';
+import { TopBarEquipmentSearchDialog } from 'components/top-bar-equipment-seach-dialog/top-bar-equipment-search-dialog';
+import SingleLineDiagramContent from './singleLineDiagram/single-line-diagram-content';
+import NetworkAreaDiagramContent from './networkAreaDiagram/network-area-diagram-content';
+const ResponsiveGridLayout = WidthProvider(Responsive);
+
+const diagramTypes = [
+    DiagramType.VOLTAGE_LEVEL,
+    DiagramType.SUBSTATION,
+    DiagramType.NETWORK_AREA_DIAGRAM,
+    DiagramType.NAD_FROM_CONFIG,
+];
+
+const styles = {
+    itemContainer: (theme: Theme) => ({
+        display: 'flex',
+        flexDirection: 'column',
+        backgroundColor: 'lightblue',
+    }),
+    diagramContainer: (theme: Theme) => ({
+        flexGrow: 1,
+        overflow: 'hidden',
+        position: 'relative',
+        backgroundColor:
+            theme.palette.mode === 'light'
+                ? theme.palette.background.paper
+                : theme.networkModificationPanel.backgroundColor,
+    }),
+    header: (theme: Theme) => ({
+        padding: theme.spacing(0.5),
+        display: 'flex',
+        alignItems: 'center',
+        backgroundColor: theme.palette.background.default,
+        borderBottom: 'solid 1px',
+        borderBottomColor: theme.palette.mode === 'light' ? theme.palette.action.selected : 'transparent',
+    }),
+};
+
+const DEFAULT_WIDTH = 1;
+const DEFAULT_HEIGHT = 2;
+
+const initialLayouts = {
+    // RGL will attempt to interpolate the rest of breakpoints based on this one
+    lg: [
+        {
+            i: 'Adder',
+            x: 0,
+            y: 0,
+            w: 1,
+            h: 1,
+        },
+    ],
+};
+
+interface DiagramLayoutProps {
+    studyUuid: UUID;
+    showInSpreadsheet: (equipment: { equipmentId: string | null; equipmentType: EquipmentType | null }) => void;
+    visible: boolean;
+}
+
+function DiagramLayout({ studyUuid, showInSpreadsheet, visible }: DiagramLayoutProps) {
+    const theme = useTheme();
+    const [layouts, setLayouts] = useState<Layouts>(initialLayouts);
+    const [cols, setCols] = useState<number>(4);
```

Comment:
Maybe relative to the mode used ? (full width or half size)

Reply:
removed